### PR TITLE
review drag drop logic

### PR DESCRIPTION
**Pull request by stklauz**
### **PR Type**
Tests, Enhancement

___

### **Description**
- Add comprehensive drag-drop integration tests covering visual feedback, reordering, and accessibility

- Extend drag-drop unit tests with critical edge cases for indentation and nested structures

- Pass `id` prop to `TodoRow` components for improved drag event handling

- Update `onDragLeave` handler signature to accept todo `id` parameter

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["TodoList Component"] -- "passes id prop" --> B["TodoRow Component"]
  B -- "calls with id" --> C["onDragLeave Handler"]
  C -- "identifies todo" --> D["useDragReorder Logic"]
  D -- "tested by" --> E["Integration Tests"]
  D -- "tested by" --> F["Edge Case Unit Tests"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>drag-drop.test.tsx</strong><dd><code>Add comprehensive drag-drop integration tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/renderer/__tests__/integration/drag-drop.test.tsx

<ul><li>New integration test file with 288 lines covering drag-drop <br>functionality<br> <li> Tests visual feedback, drag handle accessibility, and ARIA labels<br> <li> Verifies draggable elements exist with proper attributes<br> <li> Tests parent-child relationships and section boundaries<br> <li> Validates drag handle state management and rapid interactions</ul>

</details>

  </td>
  <td><a href="https://github.com/stklauz/todolo/pull/49/files#diff-64fefdeceb2d6ccbd74ddb061ea0d9c3b0ad7c35bfe786d4a7c76d54aec3fd09">+288/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>useDragReorder.test.tsx</strong><dd><code>Add critical edge case tests for indentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/renderer/features/todos/hooks/__tests__/useDragReorder.test.tsx

<ul><li>Add new test suite "Indentation - Critical Edge Cases" with 8 <br>comprehensive tests<br> <li> Test maintaining child indent when moving to different parent<br> <li> Test outdenting child when becoming root level after move<br> <li> Test reordering children of same parent<br> <li> Test preserving child indents when moving parent block<br> <li> Test complex nested structures with multiple parent-child <br>relationships<br> <li> Test inserting before target parent when dropping on child<br> <li> Test target index adjustment after source removal<br> <li> Test handling todos with undefined indent property</ul>

</details>

  </td>
  <td><a href="https://github.com/stklauz/todolo/pull/49/files#diff-792ab30ffd7123615fe837309149fbea2992156bf050241dd5c271816188f893">+234/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>TodoList.tsx</strong><dd><code>Pass todo id to TodoRow components</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/renderer/features/todos/components/TodoList/TodoList.tsx

<ul><li>Add <code>id={todo.id}</code> prop to <code>TodoRow</code> components in active todos section<br> <li> Add <code>id={todo.id}</code> prop to <code>TodoRow</code> components in completed todos section<br> <li> Enables drag event handlers to identify specific todos being dragged</ul>

</details>

  </td>
  <td><a href="https://github.com/stklauz/todolo/pull/49/files#diff-eff318802d434f0b0a8077db2a67c01edd889ddda1c79df2aa184c31be7832cb">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>TodoRow.tsx</strong><dd><code>Update TodoRow to handle id in drag events</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/renderer/features/todos/components/TodoRow/TodoRow.tsx

<ul><li>Add <code>id: number</code> to <code>TodoRowProps</code> type definition<br> <li> Update <code>onDragLeave</code> handler signature from <code>() => void</code> to <code>(id: number) </code><br><code>=> void</code><br> <li> Pass <code>id</code> parameter to <code>onDragLeave</code> callback in drag event handler<br> <li> Include <code>id</code> in destructured props</ul>

</details>

  </td>
  <td><a href="https://github.com/stklauz/todolo/pull/49/files#diff-06492baf6bd41811ebbce01ad8d90a1015880fd335e85e39ee676463a057447b">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive integration tests for drag-and-drop functionality in todos, validating drag handles, ARIA labels, and accessibility attributes.
  * Expanded test coverage for indentation edge cases in drag reordering, including nested structures, parent-child relationships, and indentation preservation during moves.

* **Internal Updates**
  * Enhanced todo row component to support improved drag reordering scenarios.

### CODE REVIEW

**Comment by stklauz** [comment]
@coderabbitai help

**Comment by stklauz** [comment]
@coderabbitai

**Comment by stklauz** [comment]
## Not up to quality standards

🔴 [4 new issues (≤ 0 minor issues)](…)
🔴 [2 new security issues (≤ 0 minor issues)](…)

[**Review Pull Request in Codacy →**](…)

##
> ✨ **AI Reviewer available:** add the `codacy-review` label to get contextual insights without leaving GitHub.

**Comment by stklauz** [comment]
## Not up to quality standards

Failing checks:
🔴 [4 new issues, 2 flagged as potential false positives (≤ 0 minor issues)](https://app.codacy.com/{provider}/{org}/{repo}/pull-requests/{N}/issues)
🔴 [2 new security issues (≤ 0 minor issues)](https://app.codacy.com/{provider}/{org}/{repo}/pull-requests/{N}/issues)

[**Review Pull Request in Codacy →**](https://app.codacy.com/{provider}/{org}/{repo}/pull-requests/{N}/issues)

##
<sub>✓ **Codacy AI Reviewer** added to this PR ✨</sub>

File: src/renderer/__tests__/integration/drag-drop.test.tsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import { screen, waitFor, within } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import * as storage from '../../features/todos/api/storage';
```
Comment (stklauz) [inline src/renderer/__tests__/integration/drag-drop.test.tsx:5]: cenas

File: src/renderer/__tests__/integration/drag-drop.test.tsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+import {
+  renderAppWithDefaults,
+  setupDefaultMocks,
+  mockStorage,
```
Comment (stklauz) [inline src/renderer/__tests__/integration/drag-drop.test.tsx:9]: cenas II

**Review by stklauz — COMMENTED** [review]
cenas
